File: android/devbricksx-kotlin-annotations/src/main/java/com/dailystudio/devbricksx/annotations/ViewModel.java
package com.dailystudio.devbricksx.annotations;

import androidx.room.ForeignKey;
import androidx.room.Index;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Retention(RetentionPolicy.CLASS)
@Target(ElementType.TYPE)
public @interface ViewModel {
}

File: android/devbricksx-kotlin-annotations/src/main/java/com/dailystudio/devbricksx/annotations/ViewModel.kt
package com.dailystudio.devbricksx.annotations

import kotlin.reflect.KClass

@Retention(AnnotationRetention.SOURCE)
@Target(AnnotationTarget.CLASS)
annotation class ViewModel(val repository: KClass<*> = Void::class,
                           val `group`: String = "",
                           val packageName: String = "")